api/envoy/extensions/filters/http/grpc_field_extraction/v3/config.proto

```diff
@@ -0,0 +1,131 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_field_extraction.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_field_extraction.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_field_extraction/v3;grpc_field_extractionv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC Field Extraction]
+// gRPC Field Extraction :ref:`configuration overview
+// <config_http_filters_grpc_field_extraction>`.
+// [#extension: envoy.filters.http.grpc_field_extraction]
+
+// GrpcFieldExtractionConfig supports extracting the fields from first gRPC
+// request message no matter if it is unary or streaming and inject the result
+// as a header. It won't propagate headers until the extraction and the
+// injection done so the following filters can use it. No op if isn't a gRPC
+// method or its type cannot be found in the proto descriptor.
+//
+// The target field should be singular primitive type or repeated primitive
+// type and its value will be extracted in string format.
```

Comment:
String format using JSON encoding? How are repeated handled?

Reply:
> String format using JSON encoding?
The target fields should be primitive type(not message) already. Their string format is straightforward.

>  How are repeated handled?
They are wrapped in ListValue of StringValue. See the example https://screenshot.googleplex.com/AWUdqcDwEp5WP4P.